// ngx-mzd-timeline/esm2015/lib/components/timeline/timeline.component.js
import { Component, ContentChildren, Input, ViewEncapsulation } from '@angular/core';
import { MzdTimelineContentComponent } from '../timeline-content/timeline-content.component';
export class MzdTimelineComponent {
    constructor() {
        this.firstContentSide = 'left';
        this.alternateSide = true;
    }
    ngOnChanges(changes) {
        this.updateContent();
    }
    ngOnInit() { }
    ngAfterViewInit() {
        this.updateContent();
    }
    updateContent() {
        if (this.contents) {
            if (this.alternateSide) {
                if (this.firstContentSide === 'left') {
                    this.contents.forEach((content, index) => {
                        content.left = (index % 2 === 0);
                        content.right = (index % 2 !== 0);
                    });
                }
                else {
                    this.contents.forEach((content, index) => {
                        content.left = (index % 2 !== 0);
                        content.right = (index % 2 === 0);
                    });
                }
            }
            else {
                this.contents.forEach(content => {
                    content.left = this.firstContentSide === 'left';
                    content.right = this.firstContentSide === 'right';
                });
            }
            this.contents.forEach(content => content.noAlternate = !this.alternateSide);
        }
    }
}
MzdTimelineComponent.decorators = [
    { type: Component, args: [{
                selector: 'mzd-timeline',
                template: "<div\n  class=\"mzd-timeline-line\"\n  [ngClass]=\"{\n    'no-alternate': !alternateSide,\n    left: firstContentSide !== 'left',\n    right: firstContentSide !== 'right'\n  }\"\n></div>\n<ng-content></ng-content>\n",
                encapsulation: ViewEncapsulation.None,
                styles: ["*{box-sizing:border-box}mzd-timeline{display:block;padding:50px 0;position:relative;width:100%}mzd-timeline .mzd-timeline-line{background:grey;height:100%;left:50%;position:absolute;right:auto;top:0;width:1px}mzd-timeline .mzd-timeline-line.no-alternate.left{left:30px;right:auto}mzd-timeline .mzd-timeline-line.no-alternate.right{left:auto;right:30px}@media screen and (max-width:599px){mzd-timeline .mzd-timeline-line{left:30px;right:auto}}"]
            },] }
];
MzdTimelineComponent.ctorParameters = () => [];
MzdTimelineComponent.propDecorators = {
    firstContentSide: [{ type: Input }],
    alternateSide: [{ type: Input }],
    contents: [{ type: ContentChildren, args: [MzdTimelineContentComponent,] }]
};
